mesonbuild/modules/gnome.py

```diff
@@ -411,20 +411,44 @@ def generate_gir(self, state, args, kwargs):
         nsversion = kwargs.pop('nsversion')
         libsources = kwargs.pop('sources')
         girfile = '%s-%s.gir' % (ns, nsversion)
+        srcdir = os.path.join(state.environment.get_source_dir(), state.subdir)
+        builddir = os.path.join(state.environment.get_build_dir(), state.subdir)
         depends = [girtarget]
         gir_inc_dirs = []
 
-        scan_command = [giscanner, '@INPUT@']
+        scan_command = [giscanner]
         scan_command += pkgargs
         scan_command += ['--no-libtool', '--namespace=' + ns, '--nsversion=' + nsversion, '--warn-all',
                          '--output', '@OUTPUT@']
 
         extra_args = mesonlib.stringlistify(kwargs.pop('extra_args', []))
         scan_command += extra_args
-        scan_command += ['-I' + os.path.join(state.environment.get_source_dir(), state.subdir),
-                         '-I' + os.path.join(state.environment.get_build_dir(), state.subdir)]
+        scan_command += ['-I' + srcdir,
+                         '-I' + builddir]
         scan_command += get_include_args(girtarget.get_include_dirs())
 
+        gir_filelist_dir = os.path.join(builddir, girtarget.get_id())
```

Comment:
Hi Jussi,

Hmm, seems like I have to use get_target_private_dir_abs() so that CI will pass.  It does seem that the doxygen tests are failing due to a missing PNG... which I probably can't attempt to fix (and I think is beyond the things that this patch tries to touch).

With blessings, thank you!